examples/proxy-server/src/main/java/example/armeria/proxy/AnimationService.java

```diff
@@ -57,19 +60,14 @@ protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) throws
         final HttpResponseWriter res = HttpResponse.streaming();
         res.write(HttpHeaders.of(HttpStatus.OK)
                              .contentType(MediaType.PLAIN_TEXT_UTF_8));
-
-        ctx.blockingTaskExecutor().submit(() -> {
-            for (int i = 0; i < 1000; i++) {
-                try {
-                    Thread.sleep(pendulumDuration);
-                } catch (Exception e) {
-                    res.close(e);
-                    return;
-                }
-                res.write(HttpData.ofUtf8(frames.get(i % 4)));
-            }
-            res.close();
-        });
+        res.onDemand(() -> streamData(ctx.eventLoop(), res, 0));
         return res;
     }
+
+    private void streamData(EventLoop executor, HttpResponseWriter writer,
+                                   int frameIndex) {
+        writer.write(HttpData.ofUtf8(frames.get(frameIndex % 4)));
```

Comment:
If `frameIndex` is negative, the return value can be also negative. How about this?
```
writer.write(HttpData.ofUtf8(frames.get(frameIndex++)));
writer.onDemand(() -> executor.schedule(() -> streamData(executor, writer, frameIndex % frames.size()),
                                        pendulumDuration, TimeUnit.MILLISECONDS));
```

Reply:
Forgot that I removed the end condition. Because I can't modify the value of `frameIndex` due to the fact that the variable has to be effectively final in lamda, I changed it a little bit. Thanks!